### PR TITLE
Don't emit f-string warning for tuple variables

### LINKED ISSUE #8296: consider-using-f-string is triggered when the formatting arguments are implicit
### Bug description

the following code should produce no diagnostics:
```python
"https://github.com/PyCQA/pylint/issues/8296"

date = (2020,1,2,3,4,5)
print("%4d-%02d-%02d %02d:%02d:%02d" % date)

def test_fn(pa_l):
    "my doc"
    return ["Alpha: {alpha:.0%}, Power: {power:.0%}, Nobs={nobs:.1f}".format(**pa)
            for pa in pa_l]
```

### Configuration

_No response_

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
************* Module a
a.py:4: [C0209(consider-using-f-string), ] Formatting a regular string which could be a f-string
a.py:8: [C0209(consider-using-f-string), test_fn] Formatting a regular string which could be a f-string
```

### Expected behavior

no diagnostics: it is impossible to replace these string formatting ops with f-strings.

### Pylint version

```shell
pylint 2.15.10
astroid 2.13.5
Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]
```

### OS / Environment

ubuntu

### Additional dependencies

_No response_

### PR DESCRIPTION
**Pull request by nickdrozd**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Description

Don't emit f-string warning for tuple variables.

Closes #8296

### CODE REVIEW

File: pylint/checkers/refactoring/recommendation_checker.py [hunk 1]
```diff
@@ -427,6 +424,11 @@ def _detect_replacable_format_call(self, node: nodes.Const) -> None:
             elif isinstance(inferred_right, nodes.List):
                 if len(inferred_right.elts) > 1:
                     return
+            # If the inferred value is a tuple and the value comes
```
Comment (nickdrozd) [inline pylint/checkers/refactoring/recommendation_checker.py:427]: This part is the actual logic change. Everything is just using variables so we don't have to call `\.parent` over and over and have ugly line breaks.

File: doc/whatsnew/fragments/8292.false_positive [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Don't emit `consider-using-f-string` for `%` with tuple variables
```
Comment (mbyrnepr2) [inline doc/whatsnew/fragments/8292.false_positive:1]:
```suggestion
Don't emit ``consider-using-f-string`` for ``%`` with tuple variables.
```

Double: code. Single: italic.

File: doc/whatsnew/fragments/8292.false_positive [hunk 3]
```diff
@@ -0,0 +1,3 @@
...
+
+Fixes #8296
```
[the remark this answers is not in the document]
Comment (mbyrnepr2) [reply doc/whatsnew/fragments/8292.false_positive:3]: The regex allows “Fixes part of” but I think the intention is “Closes”.

**Comment by Pierre-Sassoulas** [comment]
It would be nice to also handle #8224 at the same time.

**Comment by nickdrozd** [comment]
@Pierre-Sassoulas Whether these suggestions are appropriate or not is really a matter of opinion. In my opinion, the suggestion in that ticket is good, and the flagged code is strange. So I don't think I will make any attempts to "fix" it. But it someone else wanted to, I wouldn't object. Either way, I added the examples as test cases.

**Comment by nickdrozd** [comment]
The primer results all look like cases where it would be annoying to get that warning, so the change seems good.

File: pylint/checkers/refactoring/recommendation_checker.py [hunk 4]
```diff
@@ -357,37 +357,34 @@ def visit_const(self, node: nodes.Const) -> None:
             ):
                 self._detect_replacable_format_call(node)
 
+    # pylint: disable = too-many-branches
```
Comment (Pierre-Sassoulas) [inline pylint/checkers/refactoring/recommendation_checker.py:360]: We need to refactor first I think. This function became too big to understand easily. (It's also better to do the ``parent = node.parent`` refactor as a standalone so we can revert and understand the change easily).

File: pylint/checkers/refactoring/recommendation_checker.py [hunk 5]
```diff
@@ -427,6 +424,11 @@ def _detect_replacable_format_call(self, node: nodes.Const) -> None:
...
                 if len(inferred_right.elts) > 1:
                     return
+            # If the inferred value is a tuple and the value comes
+            # from a variable, don't emit the warning
+            elif isinstance(inferred_right, nodes.Tuple):
+                if isinstance(parent.right, nodes.Name):
+                    return
 
             # If all tests pass, then raise message
             self.add_message(
```
Comment (Pierre-Sassoulas) [inline pylint/checkers/refactoring/recommendation_checker.py:434]: We can add the confidence while we're at it in the previous refactor.

**Review by Pierre-Sassoulas — COMMENTED** [review]
I think we should add an option to disable this (like ``--fstrings-no-exception`` with a default to False ?). It's a convention message after all.

**Comment by nickdrozd** [comment]
Upon reflection, I don't care enough about this issue to work on it any further. Anyone who wants to take it over, feel free.

**Pull request closed without merging**